### PR TITLE
Add missing types docs

### PR DESCRIPTION
**Pull request by tpdorsey**
First pass a filling in some of the empty types docs.

## Checklist

* [ ] Unit tests for the changes exist
* [ ] Tests pass on CI and coverage remains at 100%
* [ ] Documentation reflects the changes where applicable
* [ ] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [ ] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

Selected Reviewer: @dmontagu

### CODE REVIEW

File: docs/usage/types/booleans.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 description: Support for Boolean types.
 ---
 
+`bool`
+: see below for details on how bools are validated and what values are permitted
```
Comment (dmontagu) [inline docs/usage/types/booleans.md:6]: Here's an example of what I mean in the review note

File: docs/usage/types/callables.md [hunk 2]
```diff
@@ -2,6 +2,9 @@
 description: Support for callable types.
 ---
 
+`typing.Callable`
+: see below for more detail on parsing and validation
```
Comment (dmontagu) [inline docs/usage/types/callables.md:6]: Another example

**Review by dmontagu — APPROVED** [review]
Is this being used to auto-generate something? It seems various things were added that say "see above" or "see below" and the thing to "see" is immediately next to the added comment. If this is being done to produce anchor links or otherwise make the documentation behave better, that makes sense to me, just want to make sure. (If not, it might still be reasonable, but then I'd look into the existing organization more deeply.)

The new stuff looks good to me.

File: docs/usage/types/callables.md (hunk 2 above)
Comment (tpdorsey) [reply docs/usage/types/callables.md:6]: These are artifacts from the old docs that need to get fixed in a separate editorial pass. In this pass I was just trying to get all the correct page content set up.

**Pull request merged**